test/test.go

```diff
@@ -248,9 +281,15 @@ func NewClientApplicationServer(ctx context.Context, opts ...ClientApplicationSe
 		defer wg.Done()
 		_ = d.Serve()
 	}()
+	remoteProvisioningCfg := cfg.RemoteProvisioningCfg
+	if err := remoteProvisioningCfg.Validate(); err != nil {
+		return nil, nil, err
```

Comment:
Probably 

```_ = d.Close()```

is missing here for corectness.

Reply:
fixed